projects/discussion-ui/src/lib/components/tag-all-discussion/tag-all-discussion.component.ts

```diff
@@ -0,0 +1,138 @@
+  paginationData!: any
+  fetchNewData = false
+  paramsSubscription: Subscription;
+  getParams: any;
+  cIds: any;
+
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    private discussService: DiscussionService, public activatedRoute: ActivatedRoute,
+    private configService: ConfigService,
+    private discussUtils: DiscussUtilsService
+  ) { }
+
+  ngOnInit() {
+
+    this.getParams = this.configService.getConfig()
+    this.cIds = _.get(this.getParams, 'categories')
+
+    this.activatedRoute.queryParams.subscribe((params) => {
+      this.tagName = params.tagname
+    })
+    if (this.cIds.result.length) {
+      this.fetchContextBasedTagDetails(this.tagName, this.cIds, this.currentActivePage)
+    } else {
+      this.currentActivePage = 1
+      this.fetchSingleTagDetails(this.tagName, this.currentActivePage)
+    }
+
+  }
+
+  fetchSingleTagDetails(tagname: string, page: any) {
+    this.fetchSingleCategoryLoader = true
+    this.discussService.getTagBasedDiscussion(tagname).subscribe(
+      (data: NSDiscussData.IDiscussionData) => {
+        this.similarPosts = data.topics
+        this.paginationData = data.pagination
+        this.fetchSingleCategoryLoader = false
+        this.setPagination()
+      },
+      (err: any) => {
+        // this.openSnackbar(err.error.message.split('|')[1] || this.defaultError)
+        this.fetchSingleCategoryLoader = false
+      })
+  }
+
+  fetchContextBasedTagDetails(tagname: string, cid: any, page: any) {
+    this.fetchSingleCategoryLoader = true
+    const req = {
+      request: {
+        cid: cid.result,
+        tag: tagname
+      }
+    };
+
+    this.discussService.getContextBasedTagDiscussion(req).subscribe(
+      (data: NSDiscussData.IDiscussionData) => {
+        this.similarPosts = data.result
+        // this.paginationData = data.pagination
+        this.fetchSingleCategoryLoader = false
+        this.setPagination()
+      },
+      (err: any) => {
+        // this.openSnackbar(err.error.message.split('|')[1] || this.defaultError)
+        this.fetchSingleCategoryLoader = false
+      })
+  }
+
+
+  // TODO : for pagination
+  // getNextData(tagname: string, page: any) {
+  //   return this.discussService.fetchNextTagD(tagname, page).subscribe(
+  //     (data: any) => {
+  //       this.paginationData = data.pagination
+  //       this.setPagination()
+  //       this.similarPosts = _.get(data, 'topics')
+  //     })
+  // }
+
+  setPagination() {
+    this.pager = {
+      startIndex: this.paginationData.first.page,
+      endIndex: this.paginationData.last.page,
+      pages: this.paginationData.pages,
+      currentPage: this.paginationData.currentPage,
+      totalPage: this.paginationData.pageCount,
+    }
+  }
+
+  navigateWithPage(page: any) {
+    if (page !== this.currentActivePage) {
+      this.fetchNewData = true
+
+      let routerSlug = this.configService.getConfig().routerSlug ? this.configService.getConfig().routerSlug : ''
```

Comment:
Dont add this check here. Let configService to have a check & every component should able to user the configService property directly. So that it will avoid code duplication in every component.

ex:
In ConfigService 
```
getRouterSlug() {
 return this._config.routerSlug ?this._config.routerSlug : '';
}
```

In this component we can use as below
```
this.router.navigate([`${this.configService.getRouterSlug()}${CONSTANTS.ROUTES.TAG}tag-discussions`], { queryParams: { page, tagname: this.queryParam } });
```

Reply:
Added helper method in config file